### ISSUE TITLE
dirichlet process mixtures for density estimation

### ISSUE THREAD

**Issue by OriolAbril** [pymc-devs/pymc-examples#101]
File: https://github.com/pymc-devs/pymc-examples/blob/main/examples/mixture_models/dp_mix.ipynb
Reviewers:

> The sections below may still be pending. If so, the issue is still available, it simply doesn't
> have specific guidance yet. Please refer to [this overview of updates](https://github.com/pymc-devs/pymc-examples/wiki/Notebook-updates-overview)

## Known changes needed
Changes listed in this section should all be done at some point in order to get this
notebook to a "Best Practices" state. However, these are probably not enough!
Make sure to thoroughly review the notebook and search for other updates.

### General updates
*

### ArviZ related
*

## Changes for discussion
Changes listed in this section are up for discussion, these are ideas on how to improve
the notebook but may not have a clear implementation, or fix some know issue only partially.

### General updates
*

### ArviZ related
*

**Comment by OriolAbril** [comment]
I tried moving from nbsphinx to myst-nb and was unable to do so because the widgets in this notebook are not saved correctly, see https://github.com/jupyter/jupyter-sphinx/blob/master/jupyter_sphinx/ast.py#L607. After manually editing the metadata to remove the widgets completely I was able to build the documentation.

Side note: the DE-MCMC notebooks _also_ have widgets but they are correctly saved are are not a problem.

**Comment by chiral-carbon** [comment]
can I work on this, since the previous PR only made metadata changes?

**Comment by OriolAbril** [comment]
Yes,  it still needs to be updated. I only fixed the metadata because it was broken and was blocking the switch from nbsphinx to myst-nb

**Comment by LegrandNico** [comment]
Hi, I would like to update this notebook to pymc v4, is there anything I should know?
